chia/types/blockchain_format/foliage.py

```diff
@@ -1,61 +1,5 @@
 from __future__ import annotations
 
-from dataclasses import dataclass
-from typing import List, Optional
+from chia_rs import Foliage, FoliageBlockData, FoliageTransactionBlock, TransactionsInfo
 
-from chia_rs import G2Element
-
-from chia.types.blockchain_format.coin import Coin
-from chia.types.blockchain_format.pool_target import PoolTarget
-from chia.types.blockchain_format.sized_bytes import bytes32
-from chia.util.ints import uint64
-from chia.util.streamable import Streamable, streamable
-
-
-@streamable
-@dataclass(frozen=True)
-class TransactionsInfo(Streamable):
-    # Information that goes along with each transaction block
-    generator_root: bytes32  # sha256 of the block generator in this block
-    generator_refs_root: bytes32  # sha256 of the concatenation of the generator ref list entries
-    aggregated_signature: G2Element
-    fees: uint64  # This only includes user fees, not block rewards
-    cost: uint64  # This is the total cost of this block, including CLVM cost, cost of program size and conditions
-    reward_claims_incorporated: List[Coin]  # These can be in any order
-
-
-@streamable
-@dataclass(frozen=True)
-class FoliageTransactionBlock(Streamable):
-    # Information that goes along with each transaction block that is relevant for light clients
-    prev_transaction_block_hash: bytes32
-    timestamp: uint64
-    filter_hash: bytes32
-    additions_root: bytes32
-    removals_root: bytes32
-    transactions_info_hash: bytes32
-
-
-@streamable
-@dataclass(frozen=True)
-class FoliageBlockData(Streamable):
-    # Part of the block that is signed by the plot key
-    unfinished_reward_block_hash: bytes32
-    pool_target: PoolTarget
-    pool_signature: Optional[G2Element]  # Iff ProofOfSpace has a pool pk
-    farmer_reward_puzzle_hash: bytes32
-    extension_data: bytes32  # Used for future updates. Can be any 32 byte value initially
-
-
-@streamable
-@dataclass(frozen=True)
-class Foliage(Streamable):
-    # The entire foliage block, containing signature and the unsigned back pointer
-    # The hash of this is the "header hash". Note that for unfinished blocks, the prev_block_hash
-    # Is the prev from the signage point, and can be replaced with a more recent block
-    prev_block_hash: bytes32
-    reward_block_hash: bytes32
-    foliage_block_data: FoliageBlockData
-    foliage_block_data_signature: G2Element
-    foliage_transaction_block_hash: Optional[bytes32]
-    foliage_transaction_block_signature: Optional[G2Element]
+__all__ = ["TransactionsInfo", "FoliageTransactionBlock", "FoliageBlockData", "Foliage"]
```

Comment:
Let's add at least a comment here.  Maybe at all such locations so we can readily search for them.  Maybe even have a PR deprecating this module so we get warnings and clean up the usages?

Reply:
I would expect we'd want to keep these names around for at least a while. I suspect these are the kinds of types that are likely to be pulled out by a 3rd party. Deprecating them makes sense though, but only after we transition away from them ourselves

Reply:
A was figuring a separate PR (at some point...) which deprecated it and thus caught our own uses of it would be fine.

Feel free to close this conversation.